### PR TITLE
Send recent votes in Vote Transactions

### PR DESCRIPTION
**Pull request by sagar-solana**
#### Problem

Inconsistent Vote Transaction delivery was resulting in different vote states on each node. 

#### Summary of Changes

Always pass in the last 16(configurable) votes in every vote transaction. This makes the network much more resistant to out of order Vote Transaction delivery

### CODE REVIEW

File: core/src/locktower.rs [hunk 1]
```diff
@@ -253,6 +255,13 @@ impl Locktower {
         }
     }
 
+    pub fn recent_votes(&self) -> Vec<Vote> {
+        (0..cmp::min(self.lockouts.votes.len(), NUM_RECENT))
+            .into_iter()
+            .map(|i| Vote::new(self.lockouts.votes.get(i).unwrap().slot))
```
Comment (rob-solana) [inline core/src/locktower.rs]: votes[i] will work

File: core/src/locktower.rs [hunk 2]
```diff
@@ -6,10 +6,12 @@ use solana_runtime::bank::Bank;
 use solana_sdk::account::Account;
 use solana_sdk::pubkey::Pubkey;
 use solana_vote_api::vote_state::{Lockout, Vote, VoteState, MAX_LOCKOUT_HISTORY};
+use std::cmp;
 use std::sync::Arc;
 
 pub const VOTE_THRESHOLD_DEPTH: usize = 8;
 pub const VOTE_THRESHOLD_SIZE: f64 = 2f64 / 3f64;
+const NUM_RECENT: usize = 16;
```
Comment (mvines) [inline core/src/locktower.rs]: nit: `NUM_RECENT` what?  `NUM_RECENT_VOTES` perhaps

File: core/src/replay_stage.rs [hunk 3]
```diff
@@ -294,24 +294,27 @@ impl ReplayStage {
...
-                vec![vote_ix],
-                bank.last_blockhash(),
-            );
             if let Some(new_root) = locktower.record_vote(bank.slot()) {
                 bank_forks.write().unwrap().set_root(new_root);
                 blocktree.set_root(new_root);
                 Self::handle_new_root(&bank_forks, progress);
             }
+            // Send our last few votes along with the new one
+            let mut current_votes = vec![Vote::new(bank.slot())];
```
Comment (rob-solana) [inline core/src/replay_stage.rs]: redundant

File: programs/vote_api/src/vote_state.rs [hunk 4]
```diff
@@ -117,6 +117,11 @@ impl VoteState {
         }
     }
 
+    pub fn process_votes(&mut self, mut votes: Vec<Vote>) {
+        votes.sort_by(|l, r| l.slot.cmp(&r.slot));
+        votes.into_iter().for_each(|v| self.process_vote(&v));;
```
Comment (rob-solana) [inline programs/vote_api/src/vote_state.rs]: .iter() here

File: programs/vote_api/src/vote_state.rs [hunk 5]
```diff
@@ -117,6 +117,11 @@ impl VoteState {
         }
     }
 
+    pub fn process_votes(&mut self, mut votes: Vec<Vote>) {
```
Comment (rob-solana) [inline programs/vote_api/src/vote_state.rs]: no need to sort, so try to use a slice

**Pull request merged**